src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -9804,6 +9808,75 @@ HRESULT ProfToEEInterfaceImpl::ApplyMetaData(
     return hr;
 }
 
+// Add path to list of assemblies used by the runtime to probe assembly references
+// pAssemblyPath should be absolute path of assembly
+// Assembly name should match filename.
+HRESULT ProfToEEInterfaceImpl::AddAssemblyPath(AppDomainID appDomainId,
+    const WCHAR *pAssemblyPath)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        CANNOT_TAKE_LOCK;
+        SO_NOT_MAINLINE;
+    }
+    CONTRACTL_END;
+
+    PROFILER_TO_CLR_ENTRYPOINT_SYNC_EX(kP2EEAllowableAfterAttach, (LF_CORPROF, LL_INFO1000, "**PROF: AddAssemblyPath.\n"));
+
+    BaseDomain* pDomain = reinterpret_cast<BaseDomain*>(appDomainId);
+    if (pDomain == NULL)
+    {
+        return CORPROF_E_DATAINCOMPLETE;
+    }
+
+    HRESULT hr = S_OK;
+    EX_TRY
+    {
+        SString fileName(pAssemblyPath);
+        SString simpleName;
+        bool isNativeImage;
+
+        CLRPrivBinderCoreCLR* binderContext = pDomain->GetTPABinderContext();
+        if (binderContext == nullptr)
+        {
+            hr = CORPROF_E_NOT_YET_AVAILABLE;
+        }
+        else
+        {
+            BINDER_SPACE::ApplicationContext* appContext = binderContext->GetAppContext();
+
+            hr = BINDER_SPACE::ApplicationContext::GetAssemblySimpleName(fileName, simpleName, isNativeImage);
```

Comment:
Should the profiler authors be able to provide simple assembly name? Should we update signature for it? Here we always treat .ni files as native image, probably it should be somewhere  documented,

Reply:
I don't think we have a reason to give profilers any more flexibility than we give the host, which would mean simple name is always derived from the filename on disk. The .ni suffix was used in the past but isn't used any longer, it is probably easiest and safest to fail the call if a profiler tries to provide one.